tracer/src/Datadog.Tracer.Native/cor_profiler.cpp

```diff
@@ -3688,7 +3849,56 @@ HRESULT CorProfiler::GenerateVoidILStartupMethod(const ModuleID module_id, mdMet
     // pop the returned object
     rewriterWrapper_void.Pop();
     // return
-    rewriterWrapper_void.Return();
+    rewriterWrapper_void.CreateInstr(CEE_LEAVE_S, returnInstr);
+
+    // Step 7) Catch block
+    // catch (Exception ex)
+    // {
+    //      var message = "An error occured in the managed loader: " + ex.ToString();
+    //      var chars = message.ToCharArray();
+    // 
+    //      fixed (char* p = chars)
+    //      {
+    //          var nativeLog = (delegate* unmanaged<int, IntPtr, int, void>)0xFFFFFFFF; // Replaced with the actual address
+    //          nativeLog(3, p, chars.Length);
+    //      }
+    // }
+    auto catchBegin = rewriterWrapper_void.CallMember(object_to_string_member_ref, true);
+    rewriterWrapper_void.StLocal(6);
+    rewriterWrapper_void.LoadStr(error_token);
+    rewriterWrapper_void.LoadLocal(6);
+    rewriterWrapper_void.CallMember(string_concat_member_ref, false);
+    rewriterWrapper_void.StLocal(6);
+    rewriterWrapper_void.LoadLocal(6);
+    rewriterWrapper_void.CallMember(string_to_char_array_member_ref, true);
+    rewriterWrapper_void.StLocal(8);
+    rewriterWrapper_void.LoadLocal(8);
+    rewriterWrapper_void.CreateInstr(CEE_LDC_I4_0);
+    rewriterWrapper_void.CreateInstr(CEE_LDELEMA, char_type_ref);
+    rewriterWrapper_void.CreateInstr(CEE_CONV_U);
+    rewriterWrapper_void.StLocal(7);
+    rewriterWrapper_void.CreateInstr(CEE_LDC_I4_3);
+    rewriterWrapper_void.LoadLocal(7);
+    rewriterWrapper_void.LoadLocal(6);
+    rewriterWrapper_void.CallMember(string_get_length_member_ref, true);
+    rewriterWrapper_void.LoadInt64((INT64)&NativeLog);
+    rewriterWrapper_void.CreateInstr(CEE_CONV_I);
+    rewriterWrapper_void.CreateInstr(CEE_CALLI, log_signature_token);
+    rewriterWrapper_void.CreateInstr(CEE_LDNULL);
+    rewriterWrapper_void.StLocal(8);
+    auto catchEnd = rewriterWrapper_void.CreateInstr(CEE_LEAVE_S, returnInstr);
```

Comment:
Would love if we could get a https://sharplab.io/ kind of view of this 😄

Reply:
That's roughly this, https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AbEBLDAfAASQAIBBAWACgBvKgXyqoIHZiBhK4r4ggDitqVuxAHQAzbDAwATYgAco2AG4BDAC4xiAZzXrsYYtgB2agMwAmYgH1sW0lq0wAtsAwBPADIQV0mNMZC3CJOMGoAFhCyCsrqmmHYvsBuWtgA5tq6avrEnMLcShAJxADiofaOLu6kRtIAym4uEBhaAEJuGlrEABS5eXkA2hAArmoAusRGekqaxmoAZMQqDs6ubgAKalBovX3cgyPjsxYLSxWrtdgAXjDbgbt7w2MTUzMmC1oNwE1aG1s79/snkdzO9Pt8Ltd/twAJQ8Uw5O7EQTCOjEAD0aOIMBqxAgYmIIXCkXYIBApTU5RWVRq9UazTaHQCwgIAE4MnowFCuAUigBZbqwgimHbI3YEACsxH4lC5ogwEDAKmahiM2DU3Vlwn6AAZxpMstMVerTlT1ptbvctQBGQ4mCyLZaVNwQm6avbmPUvI3aMHNX4Wy1cfqmW1mSwfOlaF0BwP9FDjIazXj9cYmp0MmBaGOW/rihNJlM+yMZrNuoNIcY6RRGdIhBwqVKuxEA5jjMBhFRQABU8mzAN4bY7UELcmMRj8suhstlIk2bllgjL6MxbDCMDAAGtDPilbAfG5iPKfBPm7sZFoxDIVEb7WxSTY7I7Vl5j/5T30ZGARNgUCIrUvP2/X9tSXRUMAwG9LH6eodGcEQABUwj3aRjFSUYYI0JwEKQmAfFQkQAEkTBgKB5U3PxSTYCAnDkTsYAAUQQdsVBrGAumBOY0EgrjgSnd88kAn8/1AmAAEcl2AKAxCVRwRE6IwIDUKwj18N9Ay4DBcMNWA1CGKAjCZdSFKUlSKMMwMMWIWoNDkYgrVhVd1y3bB8XCTQ01WQxOlM2RjEWYgxCGcCD02IZYNkaRqJUYwRARdTiDAiCwAwU5iGg5JMJEAAlIYTGwEJ0IyuDSDkOQABEor89LYKwnK8oKjDitKiqnGioxSUbJS2D02ATBatquj4+LEqUbAoHVYwdBYsBNC+Jo0sa2rcqyBqiqwkrysq9qQE6qwCK0AAJaiIEbBSwsG8zLUk0KYDkryrAiJwTuxYYtAArTNB0vSDJlfjhFsB7jtO16QFApUkpShwFrW7LlvymBCpqkQNv6qrFth+qEfRlGto60IrG6qBerUVGjAuv7uBGsaJqMKajBm4g5og6rMrqlasZhnHWuMPGlP2gAxIL3HgqAwo0aRyYk8bRduzoAcC4KrFC8L3pUbTQm+y77nloW3CV0XwtBimuEs6yYFs8wHPB4geVkclKSdao6l9Vp2kzLpHmIIi1F+B0zncf1cRGb0PPcaMg/Vb3fYjL4/XNCPvRj8ErhgIb1JkMiVDukDjcPaRM7u/9c4zhUs86cwAPz0u7uFXPEptwpZDvMkyifalneLN2tC6fVlFeeYeLtEEuN7w1Zk4yC5j4pdTZs4hTCt8CG75TstA7DARCouQ3C6KPNm0YYoBmrikg0QtfB0Yw9AgIxB/VKbxqI3wEDvw9sVScJYTUCB5AgOQgtiMQNyM9MStUmI2WQocDyn0zJXMiwlc7jgAO6digNDJGbN4aI0yhmJcOh4GdBQHAhUIgc7xRLl+QhlcvxCTIenKuX4i7DWtrbdBrM4YhEIsRKAf9agkVGjNLQ2C4KYM4d7EivD+H6EzCIXkq916UT/jvUe/cuKJhMMmUYd8LDaMtlrXYs9zbEBQIvCCrC5FQDXkqTeSjd4mGjofY+jM3bn0zFkXuN9X4PzUE/GAL9vSaRrJ/IBP85B/wARoIBa4QFFljp0GBb1i4MJELXeKyDUFsJERwjmSNcG53wSQzo4piFfgrkkghxBilJJocBEpKSwZL1YSzLJmMuEaB4XIPhUABGZmEUtVp4iOldJ6VoWR8ilSKO3j3L0sw1EFi0dxSC09c6GNsuKUxVkYZZRgJedcWQb7IzbgedG2zdlgH2UYQ5/s3AiBfBLGBKYT4uNGNCNOgYKF3SIeUwplSGkQ1Ss0/p7M+nZR2Zpc52ADkO08oCjGwKTlgr2ZCy50L3CkjuV0dRahNFzI0SmZZ8VVnECQBsyauh6aaEJBEWQqKblsD3BoIidMZpdAAERlXUD4E6CEoAqBmpvDAUA1g8IkJpKAYzwF+Fud4Xw4rrKdl0nIVlbzLTnn3sQdlnLIqpB5Xy26bBBXCrxLgEiEqGxSruaa+V40hhKr+aNcaKpmWaAgMAAAVnszJQKsEIrORcq5ppKIMpgEy8lLKqyoRVfcMJchVbTDul9fSsoGD8UVGodsXq4U+phoxGacgLkLhiWbWyzArbprCIzMiG57XUydWGzQEaayZtETkzKAB5d1ezSTwQgNZasqRybxQKZQ4llcqwauqFiIm0BcRgDAHpPwKoomUpYua2QpkoAgA1XUzoJB1J/IyP25t2SQV9tQoo+m6guiNtSFxG9UbdjDrukgHdo667gwdTTZ1CUhyFlhS209mxz0gB7auTspAiYqB3g+voT7OjSnIck+D1CgKkMrlIZw15/0nvRmBqAoEb5KBEEMPBagKnMBQ0JVJ9DyOvpfe+8Cn663TX7vabDmNAP9t5lYDw79wiDvoTQ3gxBtQIH5uJiTkmpP8wI0YIj2A/nYGILlMBa6MjSHrrbdiQ8R4zJMLomDAlpBGCFqRipCH04fQTRrfS8Vk33GnPxRN7VZQ6R2CmugQA=